### DISCUSSION TITLE
Export list of all received stations

### DISCUSSION THREAD

**Discussion by Itsisme** [KejPi/AbracaDABra#66] [Ideas]
Hello,

Great software and verry CPU light compared to others.
Well done!!

A export function of all received stations might be a nice addon? 

Chrs..

**Comment by KejPi** [comment]
Hello, 
thanks, it am trying to optimize library as much as possible.
Exporting service list should not be a big problem in general, it depends on output format and what exactly should be exported. And I am not clear about the use-case, in other words what would it be good for?

**Reply by theJazzman42** [reply]
I'm creating a DAB+ database. CSV export allows an import into Excel, LibreOffice or any other spreadsheet software.

**Reply by KejPi** [reply]
OK, why not, CSV is feasible. Just specify the contents and I will think where to integrate this feature in UI.

**Reply by theJazzman42** [reply]
The ensemble information fit well into this.

**Comment by KejPi** [comment]
Ensemble info is more like tree structure than table that fits CSV.

**Reply by KejPi** [reply]
I can only export the information available in service list, that is:

- channel
- ensemble (short) name and ID
- service (short) name and ID

**Comment by KejPi** [comment]
I can export this only for programme services (i.e. data services are excluded), does it make sense for your use-case? 

```
SId,SCIdS,Service Label,Service ShortLabel,UEID,Frequency,Ensemble Label,Ensemble ShortLabel
0x00E0100D,0,Schwarzwaldradio,Schwarzw,0x00E22007,192352,COLOR DAB+,COLOR
0x00E01735,0,Schlager Radio,Schlager,0x00E22007,192352,COLOR DAB+,COLOR
0x00E2203B,0,Expres FM,ExpresFM,0x00E22002,188928,CRa DAB+,CRa
0x00E2203B,0,Expres FM,ExpresFM,0x00E22007,192352,COLOR DAB+,COLOR
0x00E2203B,0,Expres FM,ExpresFM,0x00E22002,215072,CRa DAB+,CRa
0x00E2210E,0,RadioZET,RadioZET,0x00E22002,188928,CRa DAB+,CRa
0x00E2210E,0,RadioZET,RadioZET,0x00E22002,215072,CRa DAB+,CRa
0x00E22203,0,Radio Impuls,Impuls,0x00E22002,188928,CRa DAB+,CRa
0x00E22203,0,Radio Impuls,Impuls,0x00E22002,215072,CRa DAB+,CRa
```

**Comment by andimik** [comment]
Jan, maybe you can get in contact with Günter from FMLIST. He could tell you which format is the best.

**Reply by KejPi** [reply]
I would say this feature is not useful to FMLIST at all. It is about to export complete service list not ensemble information.

**Comment by KejPi** [comment]
Service list can be exported in 3.0.0, CSV export of ensemble is suppported for long time and you can even upload ensemble structure to FMLIST from 3.0.0.

**Comment by andimik** [comment]
will there be a message on the console that the upload to FMLIST has been completed?

**Comment by KejPi** [comment]
It is already there, you can see messages like this in application log:
```
08:54:51.087 [I] Application: Uploading ensemble information (1992 bytes)
08:54:51.606 [I] Application: Ensemble information uploaded, thank you!
```

**Comment by andimik** [comment]
Ah, I was looking for a text containing "FMLIST".

What is the reason that the upload button is always grey, although I have changed to manual FMLIST. (sorry, I don't know the correct terms by heart)?

**Reply by KejPi** [reply]
Changing the log message is good idea, I will add FMLIST there. 
Button is grey when the information was already uploaded for given ensemble. If you disable auto upload a tun new ensemble, the buttons should be active so that you can send it manually and then is becomes gray.
